### ISSUE TITLE
Omission of middle part of ternary operator breaks on PHP < 5.3

### ISSUE THREAD

**Issue by MarkMaldaba** [pear/Mail_Mime#22]
(Apologies for logging this here, but the PEAR bug tracker seems to be broken and throws an error when I try to submit a bug.)

There are a few places in the codebase that have recently been updated to remove the middle part of a ternary operator.

For example, in mime.php, line 795 has been changed from:

```php
        return $res ? $res : true;
```

to

```php
        return $res ?: true;
```

This needlessly breaks support for php < 5.3 and should be reverted.

**Comment by till** [comment]
You shouldn’t be able to install this code on whatever version of PHP you’re running. Can you use an earlier version of PEAR maybe?

**Comment by till** [comment]
The bug is probably this:
https://github.com/pear/Mail_Mime/blob/1e7ae4e5258b6c0d385a8e76add567934245d38d/package.xml#L103

**Comment by MarkMaldaba** [comment]
I'm not sure what you mean about "shouldn't be able to install".  It installs fine.

As far as I can see, this is the only thing that breaks BC for PHP < 5.3 and was only recently introduced, so I would recommend that these few changes get reverted and 5.0 compatibility be maintained.

Raising minimum version requirements should trigger a major version number increment and be indicated in the release notes, but probably only worth doing if there's a major refactor or a need to use more modern features, not just to save a few bytes in a file.

**Comment by alecpl** [comment]
It isn't just `?:`, it's also `__DIR__`, and I don't know what else.

I decided to officially drop support for PHP < 5.3.

**Comment by MarkMaldaba** [comment]
I think it's quite reasonable to drop support for PHP < 5.3, if it is done responsibly.

To that end, I would suggest that the initial fix that you did but then reverted is reinstated and released as 1.10.10, with a note indicating that the next release will drop support for PHP < 5.3.  When you do the next proper release, it should be numbered 1.11.0 and officially drop support for older versions.  (Note that this release should be whenever there is new functionality or bug fixes to release - not just for the sake of dropping support.)

This gives a couple of important advantages:

1. You can start making BC changes immediately, even though the 1.11 release may not be for a while.
2. Your downstream users aren't stuck with a broken version that *should* work on 5.2 and below, but doesn't, as 1.10.10 resolves this.
3. The release notes clearly sign-post the change in support.
4. There is the flexibility of releasing a 1.10.11 further down the line, if there was strong demand for a bugfix to 1.10.10 that affects users who may not be able to upgrade their PHP version.  Not very likely, but good practice (changing min version should never be a points release).

This would be very little extra work in terms of repository/release management, but with great benefit to many users such as ourselves who still have to support legacy systems.  As you have already made the fixes (though you subsequently reverted them) this should not involve much coding time.  And, for reference, there is only one use of `__DIR__` and that is in a unit test.

If you think it is too much work to fix existing compatibility issues for a final courtesy release, then I would be happy to take on that role and provide the necessary PRs to resolve the code.
